### PR TITLE
[fix](localgc) check gc only cache directory

### LINKED ISSUE #15237: [Bug] non-cache directory into local gc
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

The .hdr file may enter the local gc

```
W1221 15:16:14.219202  7518 file_cache.cpp:135] List dir failed: /home/disk3/lzl/test/doris/data/be/data/645/315197/536215111/315197.hdr, error: [IO_ERROR]cannot list /home/disk3/lzl/test/doris/data/be/da
ta/645/315197/536215111/315197.hdr: Not a directory
W1221 15:16:14.219213  7518 file_cache.cpp:187] List dir failed: /home/disk3/lzl/test/doris/data/be/data/645/315197/536215111/315197.hdr, error: [IO_ERROR]cannot list /home/disk3/lzl/test/doris/data/be/da
ta/645/315197/536215111/315197.hdr: Not a directory
W1221 15:16:14.219240  7518 file_cache.cpp:135] List dir failed: /home/disk3/lzl/test/doris/data/be/data/646/315199/536215111/315199.hdr, error: [IO_ERROR]cannot list /home/disk3/lzl/test/doris/data/be/da
ta/646/315199/536215111/315199.hdr: Not a directory
```

### What You Expected?

Local cache directory only into local gc

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #15237

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/beta_rowset.cpp [hunk 1]
```diff
@@ -58,6 +58,15 @@ std::string BetaRowset::segment_cache_path(int segment_id) {
     return fmt::format("{}/{}_{}", _tablet_path, rowset_id().to_string(), segment_id);
 }
 
+// just check that the format is xxx_segmentid and segmentid is numeric
+bool BetaRowset::is_segment_cache_dir(const std::string& cache_dir) {
+    auto segment_id_pos = cache_dir.find_last_of('_') + 1;
+    if (segment_id_pos >= cache_dir.size() || segment_id_pos == 0) {
```
Comment (pengxiangyu) [inline be/src/olap/rowset/beta_rowset.cpp:64]: add comment: cache_dir.find_last_of('_') == -1 or segment_id_pos >= cache_dir.size()

**Review by pengxiangyu — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.51 seconds
 load time: 634 seconds
 storage size: 17123395061 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221221094150_clickbench_pr_66213.html

**Review by pengxiangyu — APPROVED** [review]
LGTM

**Pull request merged**